blocks/bricks/recurrent.py

```diff
@@ -300,6 +301,109 @@ def apply(self, input_=None, state=None, mask=None):
+        super(LSTM, self).__init__(**kwargs)
+        self.dim = dim
+
+    def get_dim(self, name):
+        if name == 'inputs':
+            return self.dim * 4
+        else:
+            return self.dim
+
+    def _allocate(self):
+        self.W_state = shared_floatx_zeros((self.dim, 4*self.dim),
+                                           name='W_state')
+        self.W_cell = shared_floatx_zeros((self.dim, 2*self.dim),
+                                          name='W_cell')
+        self.W_cell_to_gate = shared_floatx_zeros((self.dim, self.dim),
+                                                  name='W_cell_to_gate')
+        self.biases = shared_floatx_zeros((4*self.dim,), name='biases')
+        add_role(self.W_state, WEIGHTS)
+        add_role(self.W_cell, WEIGHTS)
+        add_role(self.W_cell_to_gate, WEIGHTS)
+        add_role(self.biases, BIASES)
+
+        self.params = [self.W_state, self.W_cell, self.W_cell_to_gate,
+                       self.biases]
+
+    def _initialize(self):
+        self.biases_init.initialize(self.biases, self.rng)
+        for w in self.params[:-1]:
+            self.weights_init.initialize(w, self.rng)
+
+    @recurrent(sequences=['inputs', 'mask'], states=['states', 'cells'],
+               contexts=[], outputs=['states', 'cells'])
+    def apply(self, inputs, states, cells, mask=None):
+        """Apply the Long Short Term Memory transition.
+
+        Parameters
+        ----------
+        states : :class:`~tensor.TensorVariable`
+            The 2 dimensional matrix of current states in the shape
+            (batch_size, features). Required for `one_step` usage.
+        cells : :class:`~tensor.TensorVariable`
+            The 2 dimensional matrix of current cells in the shape
+            (batch_size, features). Required for `one_step` usage.
+        inputs : :class:`~tensor.TensorVariable`
+            The 2 dimensional matrix of inputs in the shape (batch_size,
+            features * 4).
+        mask : :class:`~tensor.TensorVariable`
+            A 1D binary array in the shape (batch,) which is 1 if there is
+            data available, 0 if not. Assumed to be 1-s only if not given.
+
+        Returns
+        -------
+        states : :class:`~tensor.TensorVariable`
+            Next states of the network.
+        cells : :class:`~tensor.TensorVariable`
+            Next cell activations of the network.
+
+        """
+        def slice_last(x, no):
+            return x.T[no*self.dim: (no+1)*self.dim].T
+
+        activation = tensor.dot(states, self.W_state) + inputs + self.biases
+        cellW_cell = tensor.dot(cells, self.W_cell)
+        in_gate = tensor.nnet.sigmoid(slice_last(activation, 0) +
+                                      slice_last(cellW_cell, 0))
+        forget_gate = tensor.nnet.sigmoid(slice_last(activation, 1) +
+                                          slice_last(cellW_cell, 1))
+        next_cells = forget_gate * cells + \
```

Comment:
What do you guys think about not  using line breaks in Blocks?

```
The preferred way of wrapping long lines is by using Python's implied line continuation inside 
parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping
expressions in parentheses. These should be used in preference to using a backslash 
for line continuation.
```
from http://legacy.python.org/dev/peps/pep-0008/